improver/utilities/solar.py

```diff
@@ -228,10 +231,26 @@ def _create_daynight_mask(self, cube):
         if cube.coord("time") in cube.coords(dim_coords=True):
             slice_coords.insert(0, cube.coord("time"))
 
-        daynight_mask = next(cube.slices(slice_coords)).copy()
-        daynight_mask.rename("day_night_mask")
-        daynight_mask.units = unit.Unit("1")
-        daynight_mask.data = np.ones(daynight_mask.data.shape, dtype="int") * self.night
+        template = next(cube.slices(slice_coords))
+        demoted_coords = [
+            crd
+            for crd in cube.coords(dim_coords=True)
+            if crd not in template.coords(dim_coords=True)
+        ]
+        for crd in demoted_coords:
+            template.remove_coord(crd)
+        attributes = generate_mandatory_attributes([template])
+        title_attribute = {"title": "Day-Night mask"}
+        data = np.ones(template.data.shape, dtype="int") * self.night
```

Comment:
This could use `np.full`, but possibly more importantly, I think this dtype will give a float64 integer, so we want to set this to be np.int32.

```suggestion
        data = np.full(template.data.shape, self.night, dtype=np.int32)
```

Reply:
Done.